concrete-core/src/specification/engines/glwe_ciphertext_vector_trivial_encryption.rs

```diff
@@ -0,0 +1,71 @@
+use concrete_commons::parameters::GlweSize;
+
+use crate::prelude::{GlweCiphertextCount, PlaintextVectorEntity};
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::GlweCiphertextVectorEntity;
+
+use super::engine_error;
+
+engine_error! {
+    GlweCiphertextVectorTrivialEncryptionError for GlweCiphertextVectorTrivialEncryptionEngine @
+    PlaintextCountMismatch => "The GLWE size times the number of ciphertexts must divide the \
+    plaintext count of the input vector."
+}
+
+impl<EngineError: std::error::Error> GlweCiphertextVectorTrivialEncryptionError<EngineError> {
+    /// Validates the inputs
+    pub fn perform_generic_checks<PlaintextVector>(
+        glwe_size: GlweSize,
+        glwe_ciphertext_count: GlweCiphertextCount,
+        input: &PlaintextVector,
+    ) -> Result<(), Self>
+    where
+        PlaintextVector: PlaintextVectorEntity,
+    {
+        if (input.plaintext_count().0 % (glwe_ciphertext_count.0 * glwe_size.0)) != 0 {
+            return Err(Self::PlaintextCountMismatch);
+        }
+        Ok(())
+    }
+}
+
```

Comment:
Again, here, the polynomial size should be verified instead.

Reply:
I changed it to check the glwe_ciphertext_count divides the number of input plaintexts (the result being the polynomial size).